egressgw: emit a warning rather than a fatal error when L7 proxy is enabled

jibi opened the pull request:
The egress gateway feature is only partially incompatible with
--enable-l7-proxy as traffic will not be forwarded to an egress gateway
only if there's an L7 policy selecting the same endpoint selected by an
egress gateway policy.

Because of this, just logging a warning should be sufficient as in most
cases both features can be enabled and used at the same time.

pchaigno reviewed (commented):
I think we should also change [the deployment step in the getting started guide](https://docs.cilium.io/en/latest/gettingstarted/egress-gateway/#enable-egress-gateway) to have `--set l7Proxy=false`. This is just to avoid users running into a warning after following our guide. In general, we want to avoid normalizing warnings.

julianwiedmann commented on daemon/cmd/daemon.go:
```diff
@@ -899,8 +899,8 @@ func NewDaemon(ctx context.Context, cancel context.CancelFunc, epMgr *endpointma
 		if option.Config.EnableL7Proxy {
-			return nil, nil, fmt.Errorf("egress gateway requires L7 proxy to be disabled (--%s=\"false\").",
-				option.EnableL7Proxy)
+			log.Warning("both egress gateway and L7 proxy are enabled. This is currently not fully supported: " +
```
Maybe
```suggestion
			log.Warning("both %s and %s are enabled. This is currently not fully supported: " +
			"if the same endpoint is selected both by an egress gateway and a L7 policy, endpoint traffic will not go through egress gateway.", option.EnableIPv4EgressGateway, option.EnableL7Proxy)
```

jibi replied:
updated while trying to keep it consistent with the other messages :+1:

jibi committed:
egressgw: emit a warning rather than a fatal error when L7 proxy is enabled

The egress gateway feature is only partially incompatible with
--enable-l7-proxy as traffic will not be forwarded to an egress gateway
only if there's an L7 policy selecting the same endpoint selected by an
egress gateway policy.

Because of this, just logging a warning should be sufficient as in most
cases both features can be enabled and used at the same time.

Signed-off-by: Gilberto Bertin <jibi@cilium.io>
